FIX TCBScans: workaround for 404 errors

MikeZeDev opened the pull request:
1) Website changed url (unrelated)
2) They deliberately introduce missing junk pictures in popular manga chapters. 

Since they cant be distinguished by name pattern, we use another method ;)

Fixes https://github.com/manga-download/hakuneko/issues/6375

MikeZeDev committed:
FIX TCBScans: workarourd for 404 errors

1) Website changed url (unrelated)
2) They deliberately introduce missing junk pictures in popular manga chapters. 

Since they cant be distinguished by name pattern, we use another method ;)

Sheepux commented on src/web/mjs/connectors/TCBScans.mjs:
```diff
@@ -45,7 +45,17 @@ export default class TCBScans extends Connector {
+        const data = await this.fetchDOM(request, 'picture > source');
+
+        //test if image really exists
+        const promiseArray = data.map(page=> new Promise((resolve, reject) =>{
```
This would flood the website with http requests on images.

MikeZeDev replied:
If you have another way to know if an image exist without trying to download it please tell me. 🤷 

I would add that since they deliberatly add fake additional pictures to thwart Hakuneko downloading, its only a fair game.

Its a HEAD request, not a big deal tho.

Sheepux replied:
I just meant to do it with a for loop so that the requests are sequentials and not done in parallel.

MikeZeDev committed:
requested changes

MikeZeDev committed:
changed logic for fake pages

missing pages are always at the end, so we start testing pictures backwards and stop when its good.

MikeZeDev committed:
tbcscans: fire all promises and wait
